lib/quantizeAttributes.js

```diff
@@ -0,0 +1,281 @@
+'use strict';
+var Cesium = require('cesium');
+
+var defined = Cesium.defined;
+
+var addExtensionsUsed = require('./addExtensionsUsed');
+var byteLengthForComponentType = require('./byteLengthForComponentType');
+var getAccessorByteStride = require('./getAccessorByteStride');
+var numberOfComponentsForType = require('./numberOfComponentsForType');
+var removeUnusedVertices = require('./removeUnusedVertices');
+
+module.exports = quantizeAttributes;
+
+/**
+ * Quantizes attributes in the provided glTF using the WEB3D_quantized_attributes extension.
+ * If options is undefined, all quantizable attributes will be quantized with the default options.
+ * If options.attributes is undefined, all quantizable attributes will be quantized with the provided options.
+ * If options.precision is undefined, the decodeMatrix be written in its full form.
+ *
+ * @param {Object} [gltf] A javascript object holding a glTF hierarchy.
+ * @param {Object} [options=undefined] Defines more specific quantization behavior.
+ * @param {Object} [options.attributes=undefined] Defines specific attributes to be quantized. This should be as arranged as meshId -> primitiveIndex -> [semantic_0, semantic_1, ... semantic_N]
+ * @param {Number} [options.precision=undefined] Restricts the number of decimal places in the decodeMatrix.
+ *
+ * @returns glTF with quantized attributes
+ *
+ * @example
+ * var gltf = {
+ *     accessors: {
+ *         accessor_0: {...}
+ *         accessor_1: {...}
+ *     },
+ *     meshes: {
+ *         geometry: {
+ *             name: 'Mesh',
+ *             primitives: [
+ *                 {
+ *                     attributes: {
+ *                         NORMAL: 'accessor_0',
+ *                         POSITION: 'accessor_1'
+ *                     }
+ *                 }
+ *             ]
+ *         }
+ *     }
+ * };
+ *
+ * // Quantize all valid attributes
+ * quantizeAttributes(gltf);
+ *
+ * // Quantize all valid attributes, restricting the decodeMatrix to 6 decimal places
+ * quantizeAttributes(gltf, {
+ *     precision: 6
+ * }
+ *
+ * // Quantize just the positions of a specific mesh
+ * quantizeAttributes(gltf, {
+ *     attributes: {
+ *         geometry: {
+ *             0: ['POSITION']
+ *         }
+ *     }
+ * });
+ */
+function quantizeAttributes(gltf, options) {
+    var accessors = gltf.accessors;
+    var bufferViews = gltf.bufferViews;
+    var buffers = gltf.buffers;
+    var i, j;
+    var accessorIds = {};
+    var precision = undefined;
```

Comment:
Just:
```javascript
var precision;
```
`undefined` is implicit and I believe generates a JSHint or WebStorm warning.